### PR TITLE
Remove localhost hardcode

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/18187

- Removed some localhost hardcodes in tests

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- unittests cover this completely

### CODE REVIEW

File: components/brave_wallet/browser/brave_wallet_utils.cc [hunk 1]
```diff
@@ -75,6 +75,12 @@ bool GetUseStagingInfuraEndpoint() {
   return env->HasVar("BRAVE_INFURA_STAGING");
 }
 
+#if defined(OS_WIN) || defined(OS_MAC)
+const char kGanacheLocalhostURL[] = "http://localhost:7545/";
+#else
+const char kGanacheLocalhostURL[] = "http://localhost:8545/";
+#endif
```
Comment (spylogsster) [inline components/brave_wallet/browser/brave_wallet_utils.cc:82]: @AlexeyBarabash double checked on my side Ganache v2.5.4 has 7545 port for linux too, can you take latest version and recheck it for you?

Comment (spylogsster) [reply components/brave_wallet/browser/brave_wallet_utils.cc:82]: I leave it as constant to be easier to change it everywhere if you have specific config

Comment (AlexeyBarabash) [reply components/brave_wallet/browser/brave_wallet_utils.cc]: @spylogsster 
You are right, port 8545 is for an obsolete version of Ganace (ganache-core: 2.13.2), which was installed by `npm install -g ganache-cli`.

The actual version downloaded from https://github.com/trufflesuite/ganache-ui/releases/download/v2.5.4/ganache-2.5.4-linux-x86_64.AppImage, `v2.5.4` uses port `7545` by default.

Sorry for troubles.

**Comment by AlexeyBarabash** [comment]
Maybe if it will be necessary to use different localhost address, then pull it into gn args?

**Review by AlexeyBarabash — APPROVED** [review]
++

**Review by bbondy — APPROVED** [review]

**Pull request merged**